src/devices/SoftwareSpi/SoftwareSpi.cs

```diff
@@ -113,7 +145,7 @@ public SoftwareSpi(int clk, int miso, int mosi, int cs, SpiConnectionSettings se
         /// <inheritdoc />
         public override void TransferFullDuplex(ReadOnlySpan<byte> dataToWrite, Span<byte> dataToRead)
         {
-            if (dataToRead.Length != dataToRead.Length)
+            if (dataToWrite.Length != dataToRead.Length)
```

Comment:
TransferFullDuplex should also check if we are able to write and throw otherwise

Reply:
Yes, you're right, it's a public method. I've moved the check to `TransferFullDuplex` instead.